# HG changeset patch
# User Rik <rik@octave.org>
# Date 1378830165 25200
#      Tue Sep 10 09:22:45 2013 -0700
# Node ID 3cf4250cc67ea9b64717c3d6ce5d51ac7c90ae42
# Parent  6643f21e27f30fb8a239ab1d5e14f0af13bce227
Hack a fix for changing initial window sizes (bug #39988).

* scripts/plot/private/__add_default_menu__.m: Call drawnow()
only after at least one uimenu has been added to menubar.

diff --git a/scripts/plot/private/__add_default_menu__.m b/scripts/plot/private/__add_default_menu__.m
--- a/scripts/plot/private/__add_default_menu__.m
+++ b/scripts/plot/private/__add_default_menu__.m
@@ -33,17 +33,16 @@ function __add_default_menu__ (fig)
 
   obj = findall (fig, "-depth", 1, "tag", "__default_menu__", "label", "&File");
   if (isempty (obj))
     ## FIXME: uimenu() will cause menubar to be displayed, even though property
     ##        menubar remains set at "none".  So, forcibly turn menubar status
     ##        on and then off to force figure to hide menubar.
     menubar_state = get (fig, "menubar");
     set (fig, "menubar", "figure");
-    drawnow ();
 
     __f = uimenu (fig, "label", "&File", "handlevisibility", "off",
                        "tag", "__default_menu__");
       uimenu (__f, "label", "Save &As", "callback", @save_cb);
       uimenu (__f, "label", "&Save", "callback", @save_cb);
       uimenu (__f, "label", "&Close", "callback", "close (gcf)");
 
     __e = uimenu (fig, "label", "&Edit", "handlevisibility", "off",
@@ -54,16 +53,21 @@ function __add_default_menu__ (fig)
         uimenu (gm, "label", "Pan+Zoom", "callback", @guimode_cb);
         uimenu (gm, "label", "Rotate+Zoom", "callback", @guimode_cb);
         uimenu (gm, "label", "None", "callback", @guimode_cb);
 
     __h = uimenu (fig, "label", "&Help", "handlevisibility", "off",
                        "tag", "__default_menu__");
       uimenu (__h, "label", "A&bout", "enable", "off");
 
+    ## FIXME: This drawnow () must occur after at least one menu item has
+    ##        been defined to avoid sizing issues in new figures.
+    ##        This may lead to flicker.  The real fix must be in the C++ code. 
+    drawnow ();
+
     set (fig, "menubar", menubar_state);
   endif
 
 endfunction
 
 function save_cb (h, e)
   lbl = get (gcbo, "label");
   if (strcmp (lbl, "&Save"))
